test/test_integration_cluster.rb

```diff
@@ -232,19 +260,29 @@ def clean_exit_pids(pids)
     end.compact
   end
 
-  def run_launcher(conf)
-    wait, ready = IO.pipe
-    @ios_to_close << wait << ready
-    events = Puma::Events.strings
-    events.on_booted { ready << "!" }
-
-    launcher = Puma::Launcher.new conf, :events => events
-
-    thr = Thread.new { launcher.run }
-
-    # wait for boot from `events.on_booted`
-    wait.sysread 1
+  # used with thread_run to define correct 'refused' errors
+  def thread_run_refused(unix: false)
```

Comment:
This feels like the normalization of deviance to me.

Reply:
It bothers me also.  Given that it seems to only apply to macOS and it's intermittent...

Reply:
Then let's at least only allow the deviance there. I'd prefer we just skip the test entirely on Darwin with a note, rather than allow all platforms to slide backward.